### DISCUSSION TITLE
Interactive Widgets for iOS

### DISCUSSION THREAD

**Discussion by kewitschka** [ABausG/home_widget#297] [Help and Q&A] [answered]
Since iOS 17 interactive home widgets are supported. Therefore I added a BackgroundIntent as described in this tutorial: https://medium.com/@ABausG/interactive-homescreen-widgets-with-flutter-using-home-widget-83cb0706a417

    import AppIntents
    import Foundation
    import home_widget
    
    @available(iOS 17, *)
    public struct BackgroundIntent: AppIntent {
      static public var title: LocalizedStringResource = "HomeWidget Background Intent"
    
      @Parameter(title: "Widget URI")
      var url: URL?
    
      @Parameter(title: "AppGroup")
      var appGroup: String?
    
      public init() {}
    
      public init(url: URL?, appGroup: String?) {
        self.url = url
        self.appGroup = appGroup
      }
    
      public func perform() async throws -> some IntentResult {
        await HomeWidgetBackgroundWorker.run(url: url, appGroup: appGroup!)
    
        return .result()
      }
    }
    
    /// This is required if you want to have the widget be interactive even when the app is fully suspended.
    /// Note that this will launch your App so on the Flutter side you should check for the current Lifecycle State before doing heavy tasks
    @available(iOS 17, *)
    @available(iOSApplicationExtension, unavailable)
    extension BackgroundIntent: ForegroundContinuableIntent {}

To be able to import home_widget, I added the following to the Podfile:

    target 'AllTasksWidgetExtension' do
      inherit! :search_paths
    end

When I try to build, I get the following error:
>Error (Xcode): Undefined symbol: absl::lts_20240116::SimpleAtod(absl::lts_20240116::string_view, double*)
>Error (Xcode): Undefined symbol: absl::lts_20240116::Base64Escape(absl::lts_20240116::string_view, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>*)
>Error (Xcode): Undefined symbol: absl::lts_20240116::RFC3339_full
>Error (Xcode): Undefined symbol: absl::lts_20240116::log_internal::LogMessage::OstreamView::stream()
>Error (Xcode): Undefined symbol: absl::lts_20240116::log_internal::LogMessage::OstreamView::OstreamView(absl::lts_20240116::log_internal::LogMessage::LogMessageData&)
>Error (Xcode): Undefined symbol: absl::lts_20240116::log_internal::LogMessage::OstreamView::~OstreamView()
>Error (Xcode): Undefined symbol: void absl::lts_20240116::log_internal::LogMessage::CopyToEncodedBuffer<(absl::lts_20240116::log_internal::LogMessage::StringType)0>(absl::lts_20240116::string_view)

This is my complete podfile  

    # Uncomment this line to define a global platform for your project
    # platform :ios, '15.0'
    
    # CocoaPods analytics sends network stats synchronously affecting flutter build latency.
    ENV['COCOAPODS_DISABLE_STATS'] = 'true'
    
    project 'Runner', {
      'Debug' => :debug,
      'Profile' => :release,
      'Release' => :release,
    }
    
    def flutter_root
      generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
      unless File.exist?(generated_xcode_build_settings_path)
        raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
      end
    
      File.foreach(generated_xcode_build_settings_path) do |line|
        matches = line.match(/FLUTTER_ROOT\=(.*)/)
        return matches[1].strip if matches
      end
      raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
    end
    
    require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)
    
    flutter_ios_podfile_setup
    target 'Runner' do
      pod 'FirebaseFirestore', :git => 'https://github.com/invertase/firestore-ios-sdk-frameworks.git', :tag => '11.0.0'
      use_frameworks!
      use_modular_headers!
    
      flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
      target 'RunnerTests' do
        inherit! :search_paths
      end
        target 'AllTasksWidgetExtension' do
          inherit! :search_paths
        end
    end
    
    post_install do |installer|
      installer.pods_project.targets.each do |target|
        flutter_additional_ios_build_settings(target)
      end
    end

**Comment by ABausG** [comment, marked as answer]
You should not use all inherit. As per the [docs](https://docs.page/abausg/home_widget/features/interactive-widgets#ios-setup):

you should only add `home_widget` to your Widget target:

```ruby
target 'YourWidgetExtension' do
   use_frameworks!
   use_modular_headers!

   pod 'home_widget', :path => '.symlinks/plugins/home_widget/ios'
end
```

**Comment by kewitschka** [comment]
Thank you very much!
